plugins/e57/io/E57Reader.cpp

```diff
@@ -1,377 +1,267 @@
+        m_rescaleFactors.clear();
+
+        // Get rescale factors for new scan, These factors are only for
+        // colors and intensity. This is required Since,
+        // 1. E57 support colors in uint8 as well as uint16. This is specified
+        //    in header as "colorLimits".
+        //		- If colorLimits is 0-255 then it is uint8 and if 0-65535 then it
+        // is uint16.
+        //		- To make this consistant, We are rescaling colors to 0-65535
+        //        range, Since default types for colors in PDAL are Unsigned16.
+        // 2. E57 supports intensity in float ranging 0-1.  This is specified in
+        //    header as "intensityLimits".
+        //		- Since default type for intensity in PDAL is Unsigned16, E57
+        //        intensity (between 0-1) need to be rescaled in uint16 (between
+        //        0-65535).
+        // To do the rescling we need the rescale factors so that we can directly
+        // multipy them with colors and intensity values.
+        // E.g. - If color limit is 0-255 then rescale factor would be 257.00
+        //        (double value) i.e 65535/(255-0)=257.
+        //		- If color limit is 0-65535 then rescale factor would be 1.00
+        //        (double value) i.e 65535/(65535-0)= 1.
+        for (auto& keyValue : m_doubleBuffers)
+        {
+            auto minmax = std::make_pair(0.0, 0.0);
+            if (e57plugin::getLimits(scan, keyValue.first, minmax))
+            {
+                m_rescaleFactors[e57plugin::e57ToPdal(keyValue.first)] =
+                    (std::numeric_limits<uint16_t>::max)() /
+                    (minmax.second - minmax.first);
+            }
+        }
 
-    point_count_t remaining = toReadCount;
-    for (point_count_t i=0; i < remaining; i++)
+        CompressedVectorNode points(scan.get("points"));
+        m_hasPose = e57plugin::getPose(scan, m_rotation, m_translation);
+        m_reader.reset(
+            new CompressedVectorReader(points.reader(m_destBuffers)));
+    }
+    catch (E57Exception& e)
     {
-        PointRef point(view->point(nextId));
-        m_currentPoint = nextId;
-        processOne(point);
-        nextId++;
+        throwError(std::to_string(e.errorCode()) + " : " + e.context());
     }
-
-    // Return what was read
-    return toReadCount;
-}
-
-bool E57Reader::processOne(PointRef& point)
-{
-    // Did we go too far?
-    if (m_currentPoint >= m_pointCount)
-        return false;
-
-    // Are we not in the same scan?
-    if (!m_chunk || !m_chunk->isInScope(m_currentPoint))
+    catch (...)
     {
-        // setup new reader
-        setupReader(m_currentPoint);
-        m_chunk->read(m_currentPoint);
+        throwError("Got an unknown exception");
     }
-
-    if (!m_chunk->isInChunk(m_currentPoint))
-        m_chunk->read(m_currentPoint);
-
-    m_chunk->setPoint(m_currentPoint,point);
-    m_currentPoint++;
-    return true;
 }
 
-QuickInfo E57Reader::inspect()
```

Comment:
You seem to have removed that function, it is important to have it to enable fast point count and bounding box evaluation of data inside an e57 file (`pdal info --summary` command)